cypress/integration/ui/filter_restrict/filter_restrict.js

```diff
@@ -0,0 +1,171 @@
+*/
+
+Given(
+    'I open an existing dashboard with non-restricted filter settings in edit mode',
+    () => {
+        toggleShowMoreButton()
+        cy.get(dashboardChipSel).contains(TEST_DASHBOARD_TITLE).click()
+        cy.get('button').contains('Edit').click()
+    }
+)
+
+When('I click to restrict filter settings', () => {
+    cy.get('[name="radio-restrict-filtering"]').parent().click()
+})
+
+When('I click away without confirming', () => {
+    cy.get('[data-test="dhis2-uicore-layer"]').click('topLeft')
+})
+
+Then('Filter Restrictions are not restricted', () => {
+    cy.get('[name="radio-allow-filtering-on-all"]').should('be.checked')
+})
+
+/*
+Scenario: I see Period and Organisation Unit if newly choosing to restrict dimensions
+*/
+
+Then(
+    'Period and Organisation Unit are displayed as selected by default',
+    () => {
+        cy.get('[data-test="dhis2-uicore-transfer-rightside"]').contains(
+            'Period'
+        )
+        cy.get('[data-test="dhis2-uicore-transfer-rightside"]').contains(
+            'Organisation Unit'
+        )
+    }
+)
+
+/*
+Scenario: I change Filter Restrictions and the changes persist while editing Filter Settings
+*/
+
+When('I add Facility Ownership to selected filters', () => {
+    cy.get('div').contains('Facility Ownership').click()
+    cy.get('[data-test="dhis2-uicore-transfer-actions-addindividual"]').click()
+})
+
+When('I click to allow all filters', () => {
+    cy.get('[name="radio-allow-filtering-on-all"]').parent().click()
+})
+
+Then(
+    'Filter Restrictions are restricted and Facility Ownership is selected',
+    () => {
+        cy.get('[name="radio-restrict-filtering"]').should('be.checked')
+        cy.get('[data-test="dhis2-uicore-transfer-rightside"]').contains(
+            'Facility Ownership'
+        )
+    }
+)
+
+/*
+Scenario: I change Filter Restrictions and the changes persist after clicking confirm
+*/
+
+When('I click Confirm', () => {
+    cy.get('button').contains('Confirm').click()
+})
+
+/*
+Scenario: I change Filter Restrictions and the changes do not persist if I click 'Exit without saving'
+*/
+
+When('I click Exit without saving', () => {
+    cy.get('button').contains('Exit without saving').click()
+})
+
+When('I open Edit mode', () => {
+    cy.get('button').contains('Edit').click()
+})
+
+/*
+Scenario: I change Filter Restrictions, save dashboard and can see the changes in filter dimensions panel
+*/
+
+When('I remove all filters from selected filters', () => {
+    cy.get('[data-test="dhis2-uicore-transfer-actions-removeall"]').click()
+})
+
+When('I save the dashboard', () => {
+    cy.get('button').contains('Save changes').click()
+})
+
+When('I click Add Filter', () => {
+    cy.contains('Add filter').click()
+})
+
+Then('I see Facility Ownership and no other dimensions', () => {
+    cy.get(filterDimensionsPanelSel).contains('Facility Ownership')
```

Comment:
This line probably isn't doing anything? If you want to assert something then you would need to use `should`

Reply:
fixed. updated `contains` lines in any Then steps have been updated to contains('whatever').should('be.visible')